components/expenses/PayoutBankInformationForm.js

```diff
@@ -232,6 +232,10 @@ const DetailsForm = ({ disabled, getFieldName, formik, host, currency }) => {
     variables: { slug: host ? host.slug : TW_API_COLLECTIVE_SLUG, currency },
   });
 
+  useEffect(() => {
```

Comment:
Can we add a comment to explain why we do that?

Reply:
Added a small comment on this one. Let me know if that make things clearer. 👍🏽

Reply:
Thanks. Based on this comment, why is it that the data is not available on initial load, but appears after a refetch?

Reply:
And why not pass it to the `requiredFieldsQuery` initially? Sorry, I'm just trying to understand why a second query is necessary

Reply:
No worries. 😄 This is what I tried initially (and [failed](https://github.com/opencollective/opencollective-frontend/pull/7138)). The problem with that is that then each time a user types something on a field `accountDetails` may change and the query reevaluated thus creating the lagging experience we saw before. 🤔

Reply:
Then it seems to me that what we need to do is to filter `accountDetails` (and eventually memorize it?) to get only the attributes that should actually trigger an update

Reply:
> Then it seems to me that what we need to do is to filter `accountDetails` (and eventually memorize it?) to get only the attributes that should actually trigger an update

I am a bit unsure of how this would work. Can you elaborate a bit please? 🙏🏽 

I think when account details are returned they come in the form of `field` groups; 

https://github.com/opencollective/opencollective-frontend/blob/c407178a16184acbe4ba95c088168c6cdca5d2b2/components/expenses/PayoutBankInformationForm.js#L41-L55

So I am not exactly sure how we could filter on just the fields that we want. The fields could depend on other fields. Like for some countries there might be `state` and some others it might not. 🤔